addons/mail/models/ir_attachment.py

```diff
@@ -124,4 +126,11 @@ def _to_store(self, store: Store, /, *, fields=None, extra_fields=None):
                     if attachment.res_model != "mail.compose.message" and attachment.res_id
                     else False
                 )
+            if "message_ids" in fields:
+                data["message_ids"] = (
+                    Store.many(attachment.message_ids.filtered(
+                        lambda m: m.model == attachment.res_model and m.res_id == attachment.res_id),
```

Comment:
Again, what we return from python should be exact fields whenever possible, so I'd return everything in here, and filter in JS if necessary.

Reply:
Other flows might need all the messages for example.

And also I want to simply the store methods and make them generic (with possible exceptions), but for that they need to all have the same shape/do the same thing, which is returning fields.
The ideal being to be able to remove them completely in favor of just using `read`/`_read_format`, and just having the Store for doing aggregate/following relations, but not to convert/format data.